File: nan21.dnet.module.md/nan21.dnet.module.md.ui.extjs/src/main/resources/webapp/js/ds/StockLocatorLovDs.js
/* 
 * DNet eBusiness Suite
 * Copyright: Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */

Ext.define("net.nan21.dnet.module.md.org.ds.model.StockLocatorLovDs", {
	extend: 'Ext.data.Model',
	fields: [
		{name:"active", type:"boolean"},
		{name:"clientId", type:"int", useNull:true},
		{name:"description", type:"string"},
		{name:"id", type:"int", useNull:true},
		{name:"name", type:"string"},
		{name:"subInventoryId", type:"int", useNull:true}
	]});

Ext.define("net.nan21.dnet.module.md.org.ds.model.StockLocatorLovDsFilter", {
	extend: 'Ext.data.Model',
	fields: [
		{name:"active", type:"boolean", useNull:true},
		{name:"description", type:"string"},
		{name:"id", type:"int", useNull:true},
		{name:"name", type:"string"},
		{name:"subInventoryId", type:"int", useNull:true},
		{name:"subInventoryId_From",type:"int", useNull:true},
		{name:"subInventoryId_To",type:"int", useNull:true}
	]
});

File: nan21.dnet.module.md/nan21.dnet.module.md.presenter/src/main/java/net/nan21/dnet/module/md/org/ds/model/StockLocatorLovDs.java
/* 
 * DNet eBusiness Suite
 * Copyright: 2010-2013 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */
package net.nan21.dnet.module.md.org.ds.model;

import net.nan21.dnet.core.api.annotation.Ds;
import net.nan21.dnet.core.api.annotation.DsField;
import net.nan21.dnet.core.api.annotation.SortField;
import net.nan21.dnet.core.presenter.model.base.AbstractTypeLov;
import net.nan21.dnet.module.md.org.domain.entity.StockLocator;

@Ds(entity = StockLocator.class, jpqlWhere = " e.active = true ", sort = {@SortField(field = StockLocatorLovDs.f_name)})
public class StockLocatorLovDs extends AbstractTypeLov<StockLocator> {

	public static final String f_subInventoryId = "subInventoryId";

	@DsField(join = "left", path = "subInventory.id")
	private Long subInventoryId;

	public StockLocatorLovDs() {
		super();
	}

	public StockLocatorLovDs(StockLocator e) {
		super(e);
	}

	public Long getSubInventoryId() {
		return this.subInventoryId;
	}

	public void setSubInventoryId(Long subInventoryId) {
		this.subInventoryId = subInventoryId;
	}
}
